app/src/main/java/com/keylesspalace/tusky/components/systemnotifications/NotificationService.kt

```diff
@@ -0,0 +1,791 @@
+            Notification.Type.FOLLOW -> "CHANNEL_FOLLOW" + account.identifier
+            Notification.Type.FOLLOW_REQUEST -> "CHANNEL_FOLLOW_REQUEST" + account.identifier
+            Notification.Type.REBLOG -> "CHANNEL_BOOST" + account.identifier
+            Notification.Type.FAVOURITE -> "CHANNEL_FAVOURITE" + account.identifier
+            Notification.Type.POLL -> "CHANNEL_POLL" + account.identifier
+            Notification.Type.SIGN_UP -> "CHANNEL_SIGN_UP" + account.identifier
+            Notification.Type.UPDATE -> "CHANNEL_UPDATES" + account.identifier
+            Notification.Type.REPORT -> "CHANNEL_REPORT" + account.identifier
+            else -> null
+        }
+    }
+
+    /**
+     * Return all active notifications, ignoring notifications that:
+     * - belong to a different account
+     * - belong to a different type
+     * - are summary notifications
+     */
+    private fun getActiveNotifications(accountId: Long, typeChannelId: String): List<StatusBarNotification> {
+        val activeNotificationsForType: MutableList<StatusBarNotification> = java.util.ArrayList()
+        for (sn in notificationManager.activeNotifications) {
+            if (sn.id != accountId.toInt()) continue
+
+            val channelId = sn.notification.group
+
+            if (channelId != typeChannelId) continue
+
+            val summaryTag = "$GROUP_SUMMARY_TAG.$channelId"
+            if (summaryTag == sn.tag) continue
+
+            activeNotificationsForType.add(sn)
+        }
+
+        return activeNotificationsForType
+    }
+
+    private fun getNotificationBuilder(notificationType: Notification.Type, account: AccountEntity, channelId: String): NotificationCompat.Builder {
+        val eventResultIntent = openNotificationIntent(context, account.id, notificationType)
+
+        val eventStackBuilder = TaskStackBuilder.create(context)
+        eventStackBuilder.addParentStack(MainActivity::class.java)
+        eventStackBuilder.addNextIntent(eventResultIntent)
+
+        val eventResultPendingIntent = eventStackBuilder.getPendingIntent(
+            account.id.toInt(),
+            pendingIntentFlags(false)
+        )
+
+        val builder = NotificationCompat.Builder(context, channelId)
+            .setSmallIcon(R.drawable.ic_notify)
+            .setContentIntent(eventResultPendingIntent)
+            .setColor(context.getColor(R.color.notification_color))
+            .setAutoCancel(true)
+            .setShortcutId(account.id.toString())
+            .setSubText(account.fullName)
+            .setVisibility(NotificationCompat.VISIBILITY_PRIVATE)
+            .setCategory(NotificationCompat.CATEGORY_SOCIAL)
+            .setOnlyAlertOnce(true)
+            .setGroup(channelId)
+            .setGroupAlertBehavior(NotificationCompat.GROUP_ALERT_SUMMARY) // Only ever alert for the summary notification
+
+        setSoundVibrationLight(account, builder)
```

Comment:
this could be an extension function for `NotificationCompat.Builder`, then it is just

```
return NotificationCompat.Builder(context, channelId)
    .something()
    ...
    .somethingElse()
```

Reply:
We want to remove that method anyway soon...